File: contest/flower.py
from typing import List
def final_size(days: List[int])->int:
    flower_size = 1
    for i in range(len(days)):
        if (i<=len(days)-2) and (days[i]==0) and (days[i+1]==0):
            return -1
        elif (i>=2) and (days[i]==1) and (days[i-1]==1):
            flower_size += 5
        elif days[i] == 1:
            flower_size += 1
    return flower_size


if __name__ == "__main__":
    t = int(input())
    for i in range(t):
        n = int(input())
        days = list(map(int, input().split()))
        print(final_size(days))

File: contest/flower.cpp
#include <bits/stdc++.h>

using namespace std;

int final_size(vector<int> days){
    int flower_size = 1;
    for(int i = 0; i < days.size(); i++){
        if(days[i]==1){
            if((i>0) && days[i-1]==1){
                flower_size+=5;
            } else {
                flower_size++;
            }
        } else{
            if(i>0 && days[i-1]==0){
                return -1;
            }
        }
    }
    return flower_size;
}

int main(){
    int t;
    cin>>t;
    for (int i = 0; i < t; i++){
        int n;
        cin>>n;
        vector<int> days(n);
        for(int j = 0; j<n; j++){
            cin>>days[j];
        }
        cout<<final_size(days)<<endl;
    }
    return 0;
}